### PR TITLE
Zero-terminate body when sending POST requests via curl

### PR DESCRIPTION
**Pull request by antkmsft**
https://github.com/Azure/azure-sdk-for-c/issues/151

### CODE REVIEW

File: sdk/core/core/src/az_curl_adapter.c [hunk 1]
```diff
@@ -162,7 +162,20 @@ az_curl_send_post_request(CURL * const p_curl, az_http_request_builder const * c
   AZ_CONTRACT_ARG_NOT_NULL(p_hrb);
 
   // Method
-  AZ_RETURN_IF_CURL_FAILED(curl_easy_setopt(p_curl, CURLOPT_POSTFIELDS, p_hrb->body.begin));
+  az_mut_span body = { 0 };
+  AZ_RETURN_IF_FAILED(az_span_malloc(p_hrb->body.size + 1, &body));
```
Comment (vhvb1989) [inline sdk/core/core/src/az_curl_adapter.c]:
```suggestion
  AZ_RETURN_IF_FAILED(az_span_malloc(p_hrb->body.size + AZ_STR_ZERO.size, &body));
```

File: sdk/core/core/src/az_curl_adapter.c [hunk 2]
```diff
@@ -162,9 +162,22 @@ az_curl_send_post_request(CURL * const p_curl, az_http_request_builder const * c
...
   // Method
-  AZ_RETURN_IF_CURL_FAILED(curl_easy_setopt(p_curl, CURLOPT_POSTFIELDS, p_hrb->body.begin));
+  az_mut_span body = { 0 };
+  AZ_RETURN_IF_FAILED(az_span_malloc(p_hrb->body.size + AZ_STR_ZERO.size, &body));
 
-  AZ_RETURN_IF_CURL_FAILED(curl_easy_perform(p_curl));
+  az_mut_span zt_buf = { 0 };
+  az_result res_code = az_mut_span_to_str(body, p_hrb->body, &zt_buf);
+
+  if (az_succeeded(res_code)) {
```
Comment (sergey-shandar) [inline sdk/core/core/src/az_curl_adapter.c]: I get it, it's like actions but more code https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/core/core/src/az_curl_slist.c#L39 ;-) Just kidding.

**Comment by sergey-shandar** [comment]
:shipit:

File: sdk/core/core/src/az_curl_adapter.c [hunk 3]
```diff
@@ -153,9 +153,22 @@ az_curl_send_post_request(CURL * const p_curl, az_http_request_builder const * c
   AZ_CONTRACT_ARG_NOT_NULL(p_hrb);
 
   // Method
-  AZ_RETURN_IF_CURL_FAILED(curl_easy_setopt(p_curl, CURLOPT_POSTFIELDS, p_hrb->body.begin));
+  az_mut_span body = { 0 };
```
Comment (antkmsft) [inline sdk/core/core/src/az_curl_adapter.c]:
```suggestion
  az_mut_span body = { 0, 0 };
```

File: sdk/core/core/src/az_curl_adapter.c [hunk 4]
```diff
@@ -153,9 +153,22 @@ az_curl_send_post_request(CURL * const p_curl, az_http_request_builder const * c
...
+  AZ_RETURN_IF_FAILED(az_span_malloc(p_hrb->body.size + 1, &body));
 
-  AZ_RETURN_IF_CURL_FAILED(curl_easy_perform(p_curl));
+  az_mut_span zt_buf = { 0 };
```
Comment (antkmsft) [inline sdk/core/core/src/az_curl_adapter.c]:
```suggestion
  az_mut_span zt_buf = { 0, 0 };
```

**Pull request merged**